### PR TITLE
working on springify the tr069 server (part 4)

### PR DESCRIPTION
**Pull request by jarlah**
with all necessary steps done, I can now port the FreeACS tr069 server to spring boot. Hurra. It will be awesome. It will be testable. And it will be working exactly the same as before :P

### CODE REVIEW

File: tr069/src/main/java/com/github/freeacs/security/SpringSecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        boolean isDigest = "digest".equalsIgnoreCase(authMethod);
+
+        if (isDigest) {
+            http.addFilter(digestAuthenticationFilter())
+                    .exceptionHandling().authenticationEntryPoint(digestEntryPoint());
+        }
+
+        ExpressionUrlAuthorizationConfigurer<HttpSecurity>.ExpressionInterceptUrlRegistry config = null;
+
+        if (isDigest || isBasic) {
```
Comment (jarlah) [inline tr069/src/main/java/com/github/freeacs/security/SpringSecurityConfig.java]: this needs more work. Have to enable anonymous if authMethod is neither digest or basic.

File: tr069/src/main/java/com/github/freeacs/tr069/SessionData.java [hunk 2]
```diff
@@ -50,7 +50,7 @@
   private String serialNumber;
 
   /** Tells whether the CPE is authenticated or not. */
-  private boolean authenticated;
+  private boolean authenticated = true;
```
Comment (jarlah) [inline tr069/src/main/java/com/github/freeacs/tr069/SessionData.java]: this field is now useless

**Comment by jarlah** [comment]
@pingwiniasty this is a pretty nasty rewrite. I don't have the patience to do small pull requests.

Your best hope to get an overview of this pr is to checkout the branch. And take a look around.

1. DBI is a bean of it self. Skipped the DBAccess wrapping.
2. Replaced custom logging with sl4j
3. Using Lombok to generate logger where I need a logger
4. Removed custom security and added conditionally loaded security config based on auth.method
etc etc

**Comment by jarlah** [comment]
Oh and I removed some tests. Need to add them back. So lets do this in two steps. First you review and do some critical thinking ;) then we merge, and then I make a new pull request with tests. I will have to rewrite the tests anyway. The old tests is safely stored in git history.

**Comment by jarlah** [comment]
Oh and I should maybe add back the feature to use or not use file auth. I can add this is in the security config based on the properties.

**Comment by pingwiniasty** [comment]
Ok, I'll check it today ;)

**Comment by jarlah** [comment]
@pingwiniasty now I have added conditional file auth, a discover unit test (WITH digest authentication, yey 👍 ). So now I shouldn't touch this more before it enters master. The test is miles more readable than the old test. I may need to add test for basic authentication and no authentication too, but the fact is that the authentication part is so loosely coupled to the tr069 controller now so that if one work, the other should also work. But yes we need to test all use cases over time.

**Comment by jarlah** [comment]
No patience:) we will do it like this. I merge now. And then you can take a look on master when ever you want @pingwiniasty. I will make sure to squash this pr into a single commit

**Comment by pingwiniasty** [comment]
Sorry, today I have no time to get to computer :(

**Comment by jarlah** [comment]
@pingwiniasty no problem :) no stress. The important thing is that you know that i am making changes. And what they are. Have a nice saturday evening!!

**Pull request merged**